src/ModelDescriber/ObjectModelDescriber.php

```diff
@@ -130,6 +130,19 @@ public function describe(Model $model, OA\Schema $schema)
             return null !== $value;
         });
 
+        // Fix for https://github.com/nelmio/NelmioApiDocBundle/issues/2222
+        // Promoted properties with a value initialized by the constructor are not considered to have a default value
+        // and are therefore not returned by ReflectionClass::getDefaultProperties(); see https://bugs.php.net/bug.php?id=81386
+        $reflClassConstructor = $reflClass->getConstructor();
+        $reflClassConstructorParameters = null !== $reflClassConstructor ? $reflClassConstructor->getParameters() : [];
+        foreach ($reflClassConstructorParameters as $parameter) {
+            if (!$parameter->isDefaultValueAvailable() || null === $parameter->getDefaultValue()) {
```

Comment:
`null` could be a valid default value, this check seems unnecessary.

```suggestion
            if (!$parameter->isDefaultValueAvailable()) {
```